server/queues/processors/NotificationsProcessor.ts

```diff
@@ -163,6 +183,28 @@ export default class NotificationsProcessor extends BaseProcessor {
         );
       }
     }
+
+    // send notifs to newly mentioned users
+    const prev = await revision.previous();
+    const oldMentions = prev ? parseMentions(prev) : [];
+    const newMentions = parseMentions(document);
+    const mentions = differenceBy(newMentions, oldMentions, "modelId");
```

Comment:
While this accounts for new mentions it doesn't account for the same user being mentioned multiple times, e.g. you have a weekly meeting minutes doc and they get mentioned one week and then the next you'd expect to get another notification.

Might be that the only change needed is to differenceBy "id" instead of "modelId"?

Reply:
Well, the `oldMentions` don't have `id` since those are obtained from the revision text and _not state_.

Reply:
Ah in that case we'll have to count the mentions per-user

Reply:
That won't tell us which is the _new_ one though I suppose, for that we'd have to look at the position in the document, if you pass the `pos` of each mention back from `parseMentions` then you can `differenceBy` that.

Reply:
That won't tell us which mention is the new one though I suppose… hmm, we could assume it's the revision creator which is fairly safe but not foolproof or store the `id` in the markdown 😬 (I actually thought we still were and only actorId was dropped)

Reply:
Let's store the `id` in markdown :smile: